docs/recipes/Partial-Rendering.md

```diff
@@ -4,18 +4,172 @@
 
 # Partial Rendering
 
+In this recipe, we will go over the setup required to start using partial rendering with One App,
+in a root Holocron module. We will use the URL to select a module and a `POST` body to populate
+the props that we want passed into that module.
+
+**Overview**
+
 Renders static markup from a Holocron module, rather than a complete page. Useful for reusing a One
 App Module's markup on a non One App web page. This will only work when dispatched on the server.
 
-```js
+**API Reference**
+
+* [`setRenderPartialOnly`](https://github.com/americanexpress/one-app-ducks#setrenderpartialonly) from [`@americanexpress/one-app-ducks`](https://github.com/americanexpress/one-app-ducks)
+
+**Prerequisites**
+
+For this recipe, One App is expected to be configured with [`ONE_ENABLE_POST_TO_MODULE_ROUTES`](../api/server/Environment-Variables.md#one_enable_post_to_module_routes) enabled and a root Holocron module configured with a partial `childRoute`.
+
+`src/components/createPartialRoute.jsx`
+```jsx
+import React from 'react';
+import PropTypes from 'prop-types';
+import { fromJS } from 'immutable';
+import ModuleRoute from 'holocron-module-route';
+import { RenderModule, composeModules } from 'holocron';
+
 import { setRenderPartialOnly } from '@americanexpress/one-app-ducks';
 
+function onPartialEnterHook(getPartialRenderProps) {
+  return ({ params: { moduleName } }, replace, cb) => {
+    // setRenderPartialOnly should only be configured on the server side
+    if (!global.BROWSER) {
+      // to kick start the partial rendering mode, the setRenderPartialOnly action
+      // is dispatched.
+      dispatch(setRenderPartialOnly(true));
+    }
+    // composeModules loads the Holocron module data and lets One App
+    // know the module was loaded.
+    dispatch(composeModules([{
+      name: moduleName,
+      props: getPartialRenderProps(),
+    }])).then(() => cb());
+  };
+}
+
+function PartialRouteComponent(props) {
+  const { params: { moduleName } } = props;
+  return (
+    <RenderModule
+      moduleName={moduleName}
+      props={props}
+    />
+  );
+}
+
+export default function createPartialRoute(getPartialRenderProps) {
```

Comment:
should we start with a simpler example of a plain `ModuleRoute` and then suggest this as a possibility if they are creating multiple partial routes.

Reply:
I agree with this point, I'll update